source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -568,28 +560,57 @@ ShaderNode* ShaderGraph::getNode(const string& name)
     return it != _nodeMap.end() ? it->second.get() : nullptr;
 }
 
-void ShaderGraph::finalize(ShaderGenerator& shadergen)
+void ShaderGraph::finalize(ShaderGenerator& shadergen, const GenOptions& options)
 {
     // Optimize the graph, removing redundant paths.
     optimize();
 
+    if (options.shaderInterfaceType == SHADER_INTERFACE_COMPLETE)
```

Comment:
Creating the shader interface during graph finalization now, by creating input sockets for all inputs to be published.